File: app/src/main/java/com/junerver/cloudnote/ui/fragment/ConfigFragment.java
package com.junerver.cloudnote.ui.fragment;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;

import com.junerver.cloudnote.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import cn.bmob.v3.BmobUser;

/**
 * A simple {@link Fragment} subclass.
 */
public class ConfigFragment extends BaseFragment {


    @BindView(R.id.ivUserAvatar)
    ImageView mIvUserAvatar;
    @BindView(R.id.btnInOut)
    Button mBtnInOut;

    @Override
    protected void init() {
        BmobUser bmobUser = BmobUser.getCurrentUser();
        if (bmobUser != null) {
            mBtnInOut.setText(R.string.login_out);
        }
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_config;
    }


    @OnClick(R.id.btnInOut)
    public void onClick() {
    }
}

File: app/src/main/java/com/junerver/cloudnote/ui/fragment/ConfigFragment.kt
package com.junerver.cloudnote.ui.fragment

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.edusoa.ideallecturer.toBean
import com.idealworkshops.idealschool.utils.SpUtils
import com.junerver.cloudnote.Constants
import com.junerver.cloudnote.R
import com.junerver.cloudnote.bean.UserInfoResp
import com.junerver.cloudnote.databinding.FragmentConfigBinding
import com.junerver.cloudnote.ui.activity.LoginActivity

/**
 * A simple [Fragment] subclass.
 */
class ConfigFragment : BaseFragment() {

    private var _binding: FragmentConfigBinding? = null

    private val binding get() = _binding!!

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentConfigBinding.inflate(inflater, container, false)
        init()
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }


    private fun init() {
        //获取用户信息
        val userInfoStr = SpUtils.decodeString(Constants.SP_USER_INFO)

        val userInfo = userInfoStr.toBean<UserInfoResp>()
        if (userInfo.username.isNotEmpty()) {
            binding.tvUserName.text = userInfo.username
        }

        binding.btnLogout.setOnClickListener {
            SpUtils.clearAll()
            startActivity(Intent(activity, LoginActivity::class.java))
            activity?.finish()
        }
    }

}